src/main/api/2/studio-api-2.yaml

```diff
@@ -2309,6 +2309,69 @@ paths:
         '500':
           $ref: '#/components/responses/InternalServerError'
 
+  /configuration/extension/file:
+    get:
+      tags:
+        - configuration
+      summary: Get a file for a given extension
+      operationId: getExtensionFile
+      parameters:
+        - name: siteId
+          in: query
+          description: The site ID
+          required: true
+          schema:
+            type: string
+        - name: module
+          in: query
+          description: The module name (e.g. studio, engine)
+          required: true
+          schema:
+            type: string
+        - name: type
+          in: query
+          description: The extension type (e.g. control, data-source)
+          required: true
+          schema:
+            type: string
+        - name: name
+          in: query
+          description: The extension name (e.g. custom-input)
+          required: true
+          schema:
+            type: string
+        - name: filename
```

Comment:
Should we also allow the caller to ask for a plugin descriptor where we get the meta-data about it and not just files? Should it be in the same API definition or add another definition?

Reply:
Decision: Future feature.